signature/plugin.go

```diff
@@ -8,33 +8,42 @@ import (
 	"fmt"
 	"time"
 
-	"github.com/notaryproject/notation-core-go/signer"
+	"github.com/notaryproject/notation-core-go/signature"
 	"github.com/notaryproject/notation-go"
 	"github.com/notaryproject/notation-go/plugin"
 )
 
 // pluginSigner signs artifacts and generates signatures.
 type pluginSigner struct {
-	runner       plugin.Runner
-	keyID        string
-	pluginConfig map[string]string
+	sigProvider       provider
+	envelopeMediaType string
+	keyID             string
+	pluginConfig      map[string]string
 }
 
-// NewSignerPlugin creates a notation.Signer that signs artifacts and generates JWS signatures
+// NewSignerPlugin creates a notation.Signer that signs artifacts and generates signatures
 // by delegating the one or more operations to the named plugin,
 // as defined in
 // https://github.com/notaryproject/notaryproject/blob/main/specs/plugin-extensibility.md#signing-interfaces.
-func NewSignerPlugin(runner plugin.Runner, keyID string, pluginConfig map[string]string) (notation.Signer, error) {
+func NewSignerPlugin(runner plugin.Runner, keyID string, pluginConfig map[string]string, envelopeMediaType string) (notation.Signer, error) {
```

Comment:
> This is called from notation cli so we will need to update notation cli before alpha-4 release.

I will discard the changes so that we don't need any update for notation-cli